agent/task_context.go

```diff
@@ -127,34 +149,41 @@ func (tc *taskContext) getTimeoutType() timeoutType {
 
 // makeTaskConfig fetches task configuration data required to run the task from the API server.
 func (a *Agent) makeTaskConfig(ctx context.Context, tc *taskContext) (*internal.TaskConfig, error) {
-	if tc.project == nil {
-		grip.Info("Fetching project config.")
-		err := a.fetchProjectConfig(ctx, tc)
-		if err != nil {
-			return nil, err
-		}
+	if tc.taskConfig != nil {
+		// This is only relevant in tests. For convenience, tests can
+		// pre-initialize a task config to use instead of fetching the task
+		// setup data using the communicator.
+		return tc.taskConfig, nil
 	}
+
+	grip.Info("Fetching project config.")
+	var err error
+	task, project, expansions, redacted, err := a.fetchProjectConfig(ctx, tc)
```

Comment:
nit: preference not to call this task since that's a package name

Reply:
This will also go away with the push.